components/compass-runtime-agent/internal/compassconnection/supervisor.go

```diff
@@ -172,14 +172,14 @@ func (s *crSupervisor) SynchronizeWithCompass(connection *v1alpha1.CompassConnec
 		s.log.Info(res)
 	}
 
-	s.log.Infof("Labeling Runtime with URLs...")
-	_, err = directorClient.SetURLsLabels(s.runtimeURLsConfig)
+	s.log.Infof("Reconciling Runtime Labels...")
+	_, err = directorClient.ReconcileLabels(s.runtimeURLsConfig)
```

Comment:
There are some failing tests: 
```
time="2021-07-28T14:07:33+02:00" level=info msg="Connection established. Director URL: https://director.com , ConnectorURL: https://cert-connector.com" Supervisor=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Processing Compass Connection, current status: Connected" CompassConnection=compass-connection Controller=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Trying to connect to Compass and apply Runtime configuration..." CompassConnection=compass-connection Controller=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Trying to maintain connection to Connector with https://cert-connector.com url..." CompassConnection=compass-connection Supervisor=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Trying to maintain certificates connection... Renewal: false" CompassConnection=compass-connection Supervisor=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Connection maintained. Director URL: https://director.com , ConnectorURL: https://cert-connector.com" CompassConnection=compass-connection Supervisor=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Reading configuration required to fetch Runtime configuration..." CompassConnection=compass-connection Supervisor=CompassConnection
time="2021-07-28T14:07:33+02:00" level=info msg="Fetching configuration from Director, from https://director.com url..." CompassConnection=compass-connection Supervisor=CompassConnection
panic: interface conversion: *mocks.DirectorClient is not director.DirectorClient: missing method ReconcileLabels

goroutine 526 [running]:
github.com/kyma-project/kyma/components/compass-runtime-agent/internal/compass/mocks.(*ClientsProvider).GetDirectorClient(0xc00049a190, 0x25473fb, 0xe, 0x0, 0x0, 0x1, 0x1, 0xc0002017d8, 0x10ad5d3)
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/internal/compass/mocks/ClientsProvider.go:74 +0x233
github.com/kyma-project/kyma/components/compass-runtime-agent/internal/compassconnection.(*crSupervisor).SynchronizeWithCompass(0xc00078e9a0, 0xc000173080, 0xc000201ab0, 0x1, 0x1)
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/internal/compassconnection/supervisor.go:143 +0x40d
github.com/kyma-project/kyma/components/compass-runtime-agent/internal/compassconnection.(*Reconciler).Reconcile(0xc000784ae0, 0x276d698, 0xc0008b68d0, 0x0, 0x0, 0xc00074e8a0, 0x12, 0xc0008b68d0, 0xc0005c0000, 0x23f5c60, ...)
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/internal/compassconnection/controller.go:118 +0x70d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001bf220, 0x276d5f0, 0xc00017e040, 0x23b4360, 0xc0005e8ca0)
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001bf220, 0x276d5f0, 0xc00017e040, 0xc0005b6200)
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc00073a010, 0xc0001bf220, 0x276d5f0, 0xc00017e040)
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/Users/I542535/GolandProjects/kyma-project/kyma/components/compass-runtime-agent/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:210 +0x425
FAIL	github.com/kyma-project/kyma/components/compass-runtime-agent/internal/compassconnection	14.557s
```

Reply:
I was missing an installation of Kubebuilder 2 and that's why this test was failing to run. Now I successfully ran it and I saw the same error. I'll fix this right now.

Reply:
Done! 🔨 
![image](https://user-images.githubusercontent.com/9533366/127469379-c56e0353-d98b-4e0d-9c12-d369f8f01c55.png)